lib/spack/spack/util/path.py

```diff
@@ -77,28 +134,37 @@ def get_system_path_max():
     return sys_max_path_length
 
 
-def get_owner_uid(path, err_msg=None):
-    if not os.path.exists(path):
-        mkdirp(path, mode=stat.S_IRWXU)
+class Path:
+    """
+    Describes the filepath separator types
+    in an enum style
+    with a helper attribute
+    exposing the path type of
+    the current platform.
+    """
+    unix = 0
+    windows = 1
+    platform_path = windows if is_windows\
+        else unix
 
-        p_stat = os.stat(path)
-        if p_stat.st_mode & stat.S_IRWXU != stat.S_IRWXU:
-            tty.error("Expected {0} to support mode {1}, but it is {2}"
-                      .format(path, stat.S_IRWXU, p_stat.st_mode))
 
-            raise OSError(errno.EACCES,
-                          err_msg.format(path, path) if err_msg else "")
-    else:
-        p_stat = os.stat(path)
+def normalize_path(path, mode=Path.unix):
```

Comment:
The term "normalize" already has an established meaning for paths (see https://docs.python.org/3/library/os.path.html#os.path.normpath) so can you pick a different name for this?